src/python/pants/backend/python/tasks/pytest_run.py

```diff
@@ -437,20 +453,29 @@ def get_pytest_rootdir():
       yield conftest, get_pytest_rootdir
 
   @contextmanager
-  def _test_runner(self, workdirs, targets, sources_map):
+  def _test_runner(self, workdirs, test_targets, sources_map):
     pytest_binary = self.context.products.get_data(PytestPrep.PytestBinary)
     with self._conftest(sources_map) as (conftest, get_pytest_rootdir):
-      with self._maybe_emit_coverage_data(workdirs, targets, pytest_binary.pex) as coverage_args:
+      with self._maybe_emit_coverage_data(workdirs,
+                                          test_targets,
+                                          pytest_binary.pex) as coverage_args:
         yield pytest_binary, [conftest] + coverage_args, get_pytest_rootdir
 
   def _do_run_tests_with_args(self, pex, args):
     try:
+      env = dict(os.environ)
+
+      # Ensure we don't leak source files or undeclared 3rdparty requirements into the py.test PEX
+      # environment.
+      pythonpath = env.pop('PYTHONPATH', None)
+      if pythonpath:
+        self.context.log.warn('scrubbed PYTHONPATH={} from py.test environment'.format(pythonpath))
```

Comment:
Presumably there is there some other scrubbing happening higher in the stack that will prevent this warning from triggering spuriously in a venv, for example? Easy to remove later.

Reply:
There is not! When you pass a custom env to pex.run it accepts you know what you're doing.